### PR TITLE
allow zero as cpu speed value in service offerings

### PR DESCRIPTION
**Pull request by DK101010**
### Description
In 4.13 it was possible to create service offerings with 0Mhz cpu speed so that se hypervisor decides how many Mhz are possible.
In 4.15 with the new UI it was not longer possible, This PR adapt the frontend check to get the same behavior  like 4.13. 
 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?
own vmware env. create service offering with 0Mhz cpu speed and build a vm.

### CODE REVIEW

File: ui/src/views/offering/AddComputeOffering.vue [hunk 1]
```diff
@@ -192,7 +192,7 @@
               rules: [{ required: true, message: $t('message.error.required.input') },
                       {
                         validator: (rule, value, callback) => {
-                          if (value && (isNaN(value) || value <= 0)) {
+                          if (value && isNaN(value)) {
```
Comment (davidjumani) [inline ui/src/views/offering/AddComputeOffering.vue:195]:
```suggestion
                          if (value && (isNaN(value) || value < 0)) {
```
Guard for negative values

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by rhtyd** [comment]
@DK101010 sounds like a bugfix - can you rebase your PR branch against 4.15?

**Comment by weizhouapache** [comment]
@DK101010 @rhtyd 
if cpu speed is 0, the used cpu capacity will be inaccurate.
it might cause problems on some vm allocation algorithms, and also cpu capacity check when start a vm.

**Comment by DK101010** [comment]
> @DK101010 @rhtyd
> if cpu speed is 0, the used cpu capacity will be inaccurate.
> it might cause problems on some vm allocation algorithms, and also cpu capacity check when start a vm.

Hi @weizhouapache, 
I'm not quite sure which impact it has, but it already works and used by us in the 4.13.  When I remind me right exists a background job that ask the hypervisor which resources are in use or is that wrong?

**Comment by mib1185** [comment]
> @DK101010 @rhtyd
> if cpu speed is 0, the used cpu capacity will be inaccurate.
> it might cause problems on some vm allocation algorithms, and also cpu capacity check when start a vm.

Hi @weizhouapache 
as I understand, CPU speed is not used for capacity calculations, but for prioritization of VMs on hypervisor, when CPU is overcommitted and CPU speed is used to select the host by its provided CPU speed - eq. serviceoffering with cpu-speed = 2000 will never deploy on host with (_base_) CPU speed of 1900, even there are enough CPU cores available.
With "CPU cap" enabled in service offering, speed will also be limited on hypervisor, even more ressources are available.
Those set CPU speed to 0 acts more as "unlimited" CPU speed usage (_independent of CPU count_)

**Review by rhtyd — APPROVED** [review]
LGTM - @davidjumani pl review/test

**Comment by davidjumani** [comment]
@blueorangutan ui

**Review by davidjumani — APPROVED** [review]

**Pull request merged**